lib/controllers/Motor.cpp

```diff
@@ -23,15 +23,14 @@ void Motor::setPower(float dutyCycle) {
   m_pwm = fmin(fabs(dutyCycle), m_limit);
 }
 
+// TODO: this function results in a warning "control reaches end of non-void function", not sure how to fix it
 Motor& Motor::operator=(int dutyCycle) {
   this->setPower(dutyCycle);
   return *this;
 }
 
 float Motor::getPower() {
-  if (m_dir) {
-    return m_pwm.read();
-  } else {
-    return -m_pwm.read();
-  }
-}
+  float temp;
```

Comment:
I don't feel like this really simplifies it much, if you want to do this you might as well do without temp and just return either case.